QuizApp/Presentation/Login/LoginViewController.swift

```diff
@@ -66,29 +87,51 @@ extension LoginViewController: ConstructViewsProtocol {
         logoLabel.font = .sourceSansPro(
             ofSize: DesignConstants.FontSize.heading.cgFloat,
             ofWeight: SourceSansProWeight.bold)
+
+        componentsStackView.spacing = 18
+        componentsStackView.axis = .vertical
+        componentsStackView.alignment = .center
+        componentsStackView.distribution = .fill
     }
 
     func defineLayoutForViews() {
+        componentsStackView.snp.makeConstraints { make in
+            make.top.leading.trailing.equalTo(view.safeAreaLayoutGuide).inset(DesignConstants.Insets.componentsInset)
+            make.bottom.equalTo(view.safeAreaLayoutGuide.snp.bottom)
+            make.centerX.equalToSuperview()
+        }
+
         logoLabel.snp.makeConstraints { make in
             make.width.equalToSuperview()
-            make.height.equalTo(DesignConstants.InputComponents.height)
-            make.top.equalTo(view.safeAreaLayoutGuide).offset(CustomConstants.logoLabelTopOffset)
             make.centerX.equalToSuperview()
         }
 
+        componentsContainerView.snp.makeConstraints { make in
+            make.leading.trailing.equalToSuperview()
+            make.top.greaterThanOrEqualToSuperview()
+            make.bottom.lessThanOrEqualToSuperview()
+        }
+
         usernameInput.snp.makeConstraints { make in
-            make.leading.trailing.equalToSuperview().inset(DesignConstants.Insets.componentsInset)
+            make.top.equalTo(componentsContainerView)
+            make.width.equalToSuperview()
```

Comment:
Why did you move from edges to width?

Reply:
looks cleaner since the insets are put once on stackview